src/core/src/lib/utils.ts

```diff
@@ -226,6 +226,10 @@ interface IObserveTarget<T> {
 }
 
 export function observeDeep(source: any, paths: string[], setFn: () => void): () => void {
+  if (source == null) {
```

Comment:
There is no issue in `observeDeep`, I think that check should be handled in `observe` function instead

Reply:
the issue is in `observeDeep` for sure.  
it is passed an undefined `source` which is passed to the `observe` and then when `unsubscribe` is called it NPEs on the if() check of `observe()`.

`observeDeep` takes `any` so code has to check for falsy values.
The fix results in a no-op unsubscribe and fixes the issue.

Reply:
@aitboudad I see you updated the PR and in essence completely removed my fix.

Although you are now handling `undefined` by creating an object the signature is still incorrect and misleading.

either `observe()` is always passed in an object or it should become `source: IObserveTarget<T> | undefined`

if not it is misleading and then this does not make sense  `if (!isObject(o)) {` as it should never happen according to the signature and TS.

Also with your fix you are still creating an empty observer which will no-op if source is indeed `undefined`. 
Why do that?

Reply:
@gkamperis `observeDeep` rely on `observe` so I think the fix should be handled inside `observe`. The solution is either to raise an error in case of passing an empty object or handle it somehow.

Could you please share the original error that produce passing an empty value to `observeDeep` so I can see if the current fix can be handled by other ways?

Reply:
FYI I've reverted this PR and instead added a check to ensure that passed value to `observeDeep` is not empty.

Reply:
@aitboudad the NPE was coming from `observeDeep(field.options, ` 
so `field.options` was `undefined`

I see you changed it so `options` is now a `path`. 
not sure how the other changes in your latest PR are related and if they will stop the original issue.

tbh idk enough about the usage of the function across the project.

since `observeDeep` is a util function for me it would be best to handle the issue within it.
you now have to make sure all callers pass a valid object.

but still `private fieldChanges(field: FormlyFieldConfigCache) {`
is contradicting with `  if (!field)  {;`

it should be `private fieldChanges(field: FormlyFieldConfigCache | undefined) {`

Reply:
> not sure how the other changes in your latest PR are related and if they will stop the original issue.

other changes not related to this.

> tbh idk enough about the usage of the function across the project.

For now only used inside `FormlyField` component

> it should be private fieldChanges(field: FormlyFieldConfigCache | undefined) {

Right, pushed a fix.

Reply:
The fix is published in https://github.com/ngx-formly/ngx-formly/releases/tag/v6.1.5